numpy/f2py/service.py

```diff
@@ -0,0 +1,327 @@
+    crackfortran.skipfuncs = crackfortran_setts["skipfuncs"]
+    crackfortran.onlyfuncs = crackfortran_setts["onlyfuncs"]
+    crackfortran.dolowercase = crackfortran_setts["do-lower"]
+
+def _set_rules(rules_setts):
+    rules.options = rules_setts
+
+def _set_capi_maps(capi_maps_setts):
+    capi_maps.load_f2cmap_file(capi_maps_setts["f2cmap"])
+    _set_additional_headers(capi_maps_setts["headers"])
+
+def _set_auxfuncs(aux_funcs_setts):
+    auxfuncs.options = {'verbose': aux_funcs_setts['verbose']}
+    auxfuncs.debugoptions = aux_funcs_setts["debug"]
+    auxfuncs.wrapfuncs = aux_funcs_setts['wrapfuncs']
+
+def _dict_append(d_out, d_in):
+    for (k, v) in d_in.items():
+        if k not in d_out:
+            d_out[k] = []
+        if isinstance(v, list):
+            d_out[k] = d_out[k] + v
+        else:
+            d_out[k].append(v)
+
+def _buildmodules(lst):
+    cfuncs.buildcfuncs()
+    outmess('Building modules...\n')
+    modules, mnames, isusedby = [], [], {}
+    for item in lst:
+        if '__user__' in item['name']:
+            cb_rules.buildcallbacks(item)
+        else:
+            if 'use' in item:
+                for u in item['use'].keys():
+                    if u not in isusedby:
+                        isusedby[u] = []
+                    isusedby[u].append(item['name'])
+            modules.append(item)
+            mnames.append(item['name'])
+    ret = {}
+    for module, name in zip(modules, mnames):
+        if name in isusedby:
+            outmess('\tSkipping module "%s" which is used by %s.\n' % (
+                name, ','.join('"%s"' % s for s in isusedby[name])))
+        else:
+            um = []
+            if 'use' in module:
+                for u in module['use'].keys():
+                    if u in isusedby and u in mnames:
+                        um.append(modules[mnames.index(u)])
+                    else:
+                        outmess(
+                            f'\tModule "{name}" uses nonexisting "{u}" '
+                            'which will be ignored.\n')
+            ret[name] = {}
+            _dict_append(ret[name], rules.buildmodule(module, um))
+    return ret
+
+
+def _generate_signature(postlist, sign_file: str):
+    outmess(f"Saving signatures to file {sign_file}" + "\n")
+    pyf = crackfortran.crack2fortran(postlist)
+    if sign_file in {"-", "stdout"}:
+        sys.stdout.write(pyf)
+    else:
+        with open(sign_file, "w") as f:
+            f.write(pyf)
+
+def _check_postlist(postlist, sign_file: str):
+    isusedby = {}
+    for plist in postlist:
+        if 'use' in plist:
+            for u in plist['use'].keys():
+                if u not in isusedby:
+                    isusedby[u] = []
+                isusedby[u].append(plist['name'])
+    for plist in postlist:
+        if plist['block'] == 'python module' and '__user__' in plist['name'] and plist['name'] in isusedby:
+            # if not quiet:
```

Comment:
Is this comment intentional?

Reply:
No, its leftover from debugging. Removing this and a few others